table/sst_file_writer.cc

```diff
@@ -266,6 +267,9 @@ Status SstFileWriter::Open(const std::string& file_path) {
   }
 
   sst_file->SetIOPriority(r->io_priority);
+  r->kv_validator = std::unique_ptr<OutputValidator>(
+      new OutputValidator(r->internal_comparator, /* enable_order_check */ true,
+                          /* enable_hash */ true));
```

Comment:
I guess this hash is unused when `!r->mutable_cf_options.paranoid_file_checks`. Maybe set enable_hash according to paranoid_file_checks?